src/mixer/playerinfo.cpp

```diff
@@ -151,16 +161,21 @@ void PlayerInfo::updateCurrentPlayingDeck() {
             maxVolume = dvol;
         }
     }
-    if (maxDeck != m_currentlyPlayingDeck) {
-        m_currentlyPlayingDeck = maxDeck;
-        locker.unlock();
+    locker.unlock();
+
+    int oldDeck = m_currentlyPlayingDeck.fetchAndStoreRelease(maxDeck);
+    if (maxDeck != oldDeck) {
         emit currentPlayingDeckChanged(maxDeck);
-        emit currentPlayingTrackChanged(getCurrentPlayingTrack());
+        TrackPointer track;
```

Comment:
This could become emit `currentPlayingTrackChanged(getCurrentPlayingTrack())` if only `getCurrentPlayingDeck()` and `getCurrentPlayingTrack()` would be idempotent and work as expected.